src/main/java/com/mongodb/spark/sql/connector/read/partitioner/FieldPartitioner.java

```diff
@@ -29,47 +28,45 @@
 import org.jetbrains.annotations.ApiStatus;
 
 import org.bson.BsonDocument;
-import org.bson.BsonNull;
 
 import com.mongodb.spark.sql.connector.config.ReadConfig;
 import com.mongodb.spark.sql.connector.exceptions.ConfigException;
 import com.mongodb.spark.sql.connector.read.MongoInputPartition;
 
 /**
- * Partitions collections using one or more fields.
+ * Partitions collections using a single field.
  *
  * <ul>
- *   <li>{@value PARTITION_FIELD_LIST_CONFIG}: A comma delimited list of fields to be used for
- *       partitioning. Defaults to: {@value ID_FIELD}.
+ *   <li>{@value PARTITION_FIELD_CONFIG}: A comma delimited list of fields to be used for
+ *       partitioning. Defaults to: {@value PARTITION_FIELD_DEFAULT}.
  * </ul>
  *
- * <p>Note: The Partitioner must provide unique partitions without any duplicates or overlapping
- * values for each field in the field list. The partition field values must also be sorted ascending
- * so that they are growing in value.
+ * <p>Note: The Partitioner must provide unique partitions without any duplicates. The partition
+ * field values must also be sorted ascending so that they are growing in value.
```

Comment:
I still don't understand the _"partition field values must also be sorted ascending"_ requirement. If it's about the `upperBounds` passed to `createMongoInputPartitions`, then we should move the requirement there in order to clarify the current _"an ordered list of the upper boundaries for each partition"_ requirement. But if not, then I can't think of how to interpret the requirement, could you please help me to understand it?

Reply:
I have removed the note as its just for `createMongoInputPartitions`  - the list must be ordered as per the javadoc because the previous partition is used as the lower bounds. I've updated it to comment the requirement not to have any duplicate partitions.